lib/screens/register_screen.dart

```diff
@@ -0,0 +1,298 @@
+                  },
+                  decoration: kTextFieldDecoration.copyWith(
+                      hintText: 'Last Name',
+                      prefixIcon: const Icon(
+                        FontAwesomeIcons.user,
+                        color: kAppColor,
+                      ))),
+              const SizedBox(height: 5.0),
+              TextFormField(
+                autocorrect: false,
+                controller: _bioController,
+                validator: (value) {
+                  if (value == null || value.isEmpty) {
+                    return 'Bio cannot be empty';
+                  }
+                  return null;
+                },
+                decoration: kTextFieldDecoration.copyWith(
+                    hintText: 'Bio',
+                    prefixIcon: const Icon(
+                      FontAwesomeIcons.pen,
+                      color: kAppColor,
+                    )),
+              ),
+              const SizedBox(height: 5.0),
+              TextFormField(
+                  autocorrect: false,
+                  controller: _hometownController,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Hometown cannot be empty';
+                    }
+                    return null;
+                  },
+                  decoration: kTextFieldDecoration.copyWith(
+                      hintText: 'Hometown',
+                      prefixIcon: const Icon(
+                        FontAwesomeIcons.home,
+                        color: kAppColor,
+                      ))),
+              const SizedBox(height: 5.0),
+              TextFormField(
+                  autocorrect: false,
+                  controller: _yearController,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Year cannot be empty';
+                    }
+                    return null;
+                  },
+                  decoration: kTextFieldDecoration.copyWith(
+                      hintText: 'Year of Birth',
+                      prefixIcon: const Icon(
+                        FontAwesomeIcons.birthdayCake,
+                        color: kAppColor,
+                      ))),
+              const SizedBox(height: 5.0),
+              SizedBox(
+                width: double.infinity,
+                child: ElevatedButton(
+                  onPressed: () async {
+                    if (_formKey.currentState!.validate()) {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                          const SnackBar(content: Text('Adding User')));
+                    }
+                    await register();
+                    Navigator.of(context).pushNamed('/login');
+                  },
+                  child: const Text("Register"),
+                ),
+              ),
+            ])));
+  }
+
+  Future<void> register() async {
```

Comment:
Consider moving this register() function to the services/firebase_service.dart folder where all other functions that use firebase are. Could help keep all firebase logic in one place.